src/libstd/thread/mod.rs

```diff
@@ -1267,6 +1267,7 @@ impl<T> JoinInner<T> {
 /// [`Clone`]: ../../std/clone/trait.Clone.html
 /// [`thread::spawn`]: fn.spawn.html
 /// [`thread::Builder::spawn`]: struct.Builder.html#method.spawn
+#[must_use]
```

Comment:
It might be a good idea to add a short message, describing the reason for `must_use`, like

```
#[must_use = "it is better to explicitly join the spawned thread, otherwise it might be terminated abruptly, without running destructors"]
```

Reply:
Done.